src/OpenSage.Game/Gui/Apt/AptRenderer.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using System.Text;
+using System.Threading.Tasks;
+using OpenSage.Data.Apt;
+using OpenSage.LowLevel.Graphics2D;
+using OpenSage.LowLevel.Graphics3D;
+
+namespace OpenSage.Gui.Apt
+{
+    public sealed class AptRenderer
+    {
+        public static void RenderGeometry(DrawingContext drawingContext,AptContext context,
+            Geometry shape, ItemTransform transform)
+        {
+            drawingContext.Transform(transform.GeometryTransform);
+            drawingContext.ColorTransform(transform.ColorTransform);
+
+            foreach (var e in shape.Entries)
+            {
+                switch (e)
+                {
+                    case GeometryLines l:
```

Comment:
Nice as the new `switch` on types is, would this be better done as a virtual `Render` method on these types? Or did you do it like this because only a few entry types are drawable?

Reply:
I didn't want to do that, since the entries are part of the Data namespace. And having Renderlogic inside the data structures didn't seem good for me